### PR TITLE
Feature/cancancan

### PR DESCRIPTION
**Pull request by carlosmuerto**

### CODE REVIEW

**Review by andym80 — APPROVED** [review]

**Review by andym80 — APPROVED** [review]

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 <main class="user-posts">
+	<p><%= can? :read, @user%></p>
+	<p><%= can? :read, @posts%></p>
```
Comment (wandergithub) [inline app/views/posts/index.html.erb]: - [x] Hey @carlosmuerto . You are printing two boolean to the screen. I think you missed deleting them after debugging.

File: Gemfile [hunk 2]
```diff
@@ -5,6 +5,9 @@ gem 'rubocop', '>= 1.0', '< 2.0'
 
 gem 'composite_primary_keys', '~>14.0'
 
+# handle Authorization
+gem 'cancancan'
+
```
Comment (wandergithub) [inline Gemfile:10]: Nice, adding comments to the Gem file is something I've never done… I'll try to overcome my laziness and start adding them. Very informative.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -6,10 +6,22 @@ def create
...
 
+  def destroy
+    @comment = Comment.includes(:author, :post).find(params[:id])
+    @post = @comment.post
+    authorize! :create, @comment
+
+    @comment.destroy
+    redirect_to user_post_path @post.author, @post
+  end
+
```
Comment (wandergithub) [inline app/controllers/comments_controller.rb]: Take a look at **line 19**.

You are checking authorization for `:create` and the action is `:destroy`.

I actually checked the CanCanCan docs to make sure the action aliases are different:
image

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -1,35 +1,41 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!
+  authorize_resource
 
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts.order(updated_at: :desc).page(params[:page])
+    @posts = @user.posts.accessible_by(current_ability).order(updated_at: :desc).page(params[:page])
```
Comment (wandergithub) [inline app/controllers/posts_controller.rb:7]: Nice use of `#accessible_by` to query just what the user can :read.

File: app/helpers/likes_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module LikesHelper
```
Comment (wandergithub) [inline app/helpers/likes_helper.rb:1]: - [ ] Warning here, empty module.

**Review by wandergithub — COMMENTED** [review]
### Reviewed
Hey @carlosmuerto, here is my review of your project. Please read the comments to know the details.

Thanks for considering me, to check your work. Happy coding! 👍🏽 
😄

**Pull request merged**